tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/Glorot.java

```diff
@@ -0,0 +1,84 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+
+package org.tensorflow.framework.initializers;
+
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.family.TNumber;
+import org.tensorflow.types.family.TType;
+
+/**
+ * The Glorot initializer, also called Xavier initializer.
+ *
+ * <p>Draws samples from a random distribution.
+ *
+ * <p>If the distribution is TRUNCATED_NORMAL, then the distribution is centered on 0 with <code>
+ *  stddev = Math.sqrt(2. / (fanIn + fanOut))</code> where <code>fanIn</code> is the number of input
+ * units in the weight tensor and <code>fanOut</code> is the number of output units in the weight
+ * tensor.
+ *
+ * <p>If the distribution is UNIFORM, then samples are drawn from a uniform distribution within
+ * <code>[-limit, limit]</code>, where <code>limit = sqrt(6 / (fanIn + fanOut))</code> ( <code>fanIn
+ * </code> is the number of input units in the weight tensor and <code> fanOut</code> is the number
+ * of output units).
+ *
+ * <p>Examples:
+ *
+ * <p>Glorot Normal:
+ *
+ * <pre>
+ *     long seed = 1001l;
+ *     Glorot&lt;TFloat32, TFloat32&gt; initializer =
+ *             new org.tensorflow.framework.initializers.Glorot&lt;&gt;(tf,
+ *             Distribution.TRUNCATED_NORMAL, seed);
+ *     Operand&lt;TFloat32&gt; values =
+ *             initializer.call(tf.constant(Shape.of(2,2)), TFloat32.DTYPE);
+ * </pre>
+ *
+ * <p>Glorot Uniform:
+ *
+ * <pre>
+ *    long seed = 1001l;
+ *    Glorot&lt;TFloat32, TFloat32&gt; initializer =
+ *             new org.tensorflow.framework.initializers.Glorot&lt;&gt;(tf,
+ *             Distribution.UNIFORM, seed);
+ *     Operand&lt;TFloat32&gt; values =
+ *             initializer.call(tf.constant(Shape.of(2,2)), TFloat32.DTYPE);
+ * </pre>
+ *
+ * @param <T> The TType for the call operation
+ * @param <U> The TNumber for the call operation
+ * @see VarianceScaling.Distribution
+ * @see <a href="http://proceedings.mlr.press/v9/glorot10a.html">Glorot et al., 2010</a>
+ */
+public class Glorot<T extends TType, U extends TNumber> extends VarianceScaling<T, U> {
+
+  public static final double SCALE = 1.0;
+
+  /**
+   * Creates a Glorot initializer
+   *
+   * @param tf the TensorFlow Ops
+   * @param distribution The distribution type for the Glorot initializer.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and dtype.
+   * @throws java.lang.IllegalArgumentException if the distribution is not TRUNCATED_NORMAL or
```

Comment:
So I understand that this is to replicate the behaviour in Keras, where GlorotNormal means use the Glorot variance and sample from a truncated normal, but why does it throw if people want to draw from a regular normal distribution? Also it looks like the super constructor doesn't throw IllegalArgumentException in this case, only if it got a zero `SCALE` param.

Reply:
In Keras, they only use Normal and Uniform. But it seems a little arbitrary. 
Here is the original discussion from the Keras thread on why they chose these two.
[Keras Issue on `GlorotNormal` vs `GlorotUniform` ](https://github.com/keras-team/keras/issues/52)

The `@throws` class was left in by mistake when I used to check for only TRUNCATED_NORMAL and UNIFORM. I will remove it.

Reply:
I think exposing all three is fine, and maybe a note to say that Keras uses TRUNCATED_NORMAL in the ones that produce normal values.

Reply:
I added class JavaDoc in all three types, `Glorot`, `He`, and `LeCun`, highlighting how to reproduce `xxxxxNormal`, and `xxxxxUniform` by using `TRUNCATED_NORMAL` and `UNIFORM`, respectively.